Refuse to load .gdbinit files in $PWD that are not owned by us
or are world writable due to security concerns.

http://sourceware.org/ml/gdb-patches/2007-03/msg00027.html

--- gdb/main.c
+++ gdb/main.c
@@ -763,7 +763,14 @@
       || memcmp ((char *) &homebuf, (char *) &cwdbuf, sizeof (struct stat)))
     if (!inhibit_gdbinit)
       {
-	catch_command_errors (source_script, gdbinit, 0, RETURN_MASK_ALL);
+	if (cwdbuf.st_uid != getuid() || cwdbuf.st_mode & S_IWOTH)
+	  {
+	    warning (_("Refusing to load unowned/world writable gdbinit `%s'.\n"), gdbinit);
+	  }
+	else
+	  {
+	    catch_command_errors (source_script, gdbinit, 0, RETURN_MASK_ALL);
+	  }
       }
 
   for (i = 0; i < ncmd; i++)
